autopush/jwt.py

```diff
@@ -0,0 +1,102 @@
+import base64
+import binascii
+import json
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives.asymmetric import ec
+from cryptography.hazmat.primitives import hashes
+
+from pyasn1.type import univ, namedtype
+from pyasn1.codec.der.encoder import encode
+
+
+"""Why hand roll?
+Most python JWT libraries either use a python elliptic curve library directly,
+or call one that does, or is abandoned, or a dozen other reasons.
+
+After spending half a day looking for reasonable replacements, I decided to
+just write the functions we need directly.
+
+THIS IS NOT A FULL JWT REPLACEMENT.
+
+"""
+
+
+def repad(string):
+    # type: (str) -> str
+    """Adds padding to strings for base64 decoding"""
+    if len(string) % 4:
+        string += '===='[len(string) % 4:]
+    return string
+
+
+class VerifyJWT:
```

Comment:
Partly. Mostly I wanted to keep the basic concept the same so I didn't have to go do a massive code rewrite.